docs/joss/paper.md

```diff
@@ -0,0 +1,217 @@
+the same steps can be executed and interacted with one-by-one in
+the REPL or in a single Jupyter or Pluto notebook.
+We thereby achieve an interactivity of a simulation and its various model components
+far beyond the options provided in a monolithic interface.
+At the same time, defaults, set to well-established test cases, 
+enable even inexperienced users to run simulations in just a few lines of code. 
+
+To be extensible with new model components, SpeedyWeather.jl relies on Julia's
+multiple dispatch programming paradigm [@Bezanson2017].
+Every model component is defined as a new type.
+For example, to define precipitation due to the physical process of large-scale condensation,
+one would define `MyCondensation` as a new subtype of `AbstractCondensation`.
+One then only needs to extend the `initialize!` and `condensation!`
+functions for this new type. Passing on `condensation = MyCondensation()`
+to the model constructor then implements this new model component without
+the need to branch off or overwrite existing model components.
+Conceptually similar scientific modelling paradigms have been very successful
+in the Python-based generic partial differential equation solver Dedalus [@Burns2020],
+the process-oriented climate model CLIMLAB [@Rose2018],
+and the Julia ocean model Oceananigans.jl [@Ramadhan2020].
+
+![Surface temperature simulated with the primitive equation model in SpeedyWeather.jl.
+(Figure will be updated) \label{fig:primitive}](primitive.jpg)
+
+The dynamical core of SpeedyWeather.jl uses established numerics
+[@Bourke1972; @Hoskins1975; @Simmons1978; @Simmons1981],
+widely adopted in numerical weather prediction. It is based on the spherical
+harmonic transform [@reinecke2013; @stompor2011] with a leapfrog-based semi-implicit
+time integration [@Hoskins1975] and a Robert-Asselin-Williams filter [@Williams2011; @Amezcua2011].
+The spherical harmonic transform is grid-flexible [@Willmert2020]. Any iso-latitude ring-based
+grid can be used and new grids can be externally defined and passed in
+as an argument. Many grids are already implemented: the conventional
+Gaussian grid, a regular longitude-latitude grid, 
+the octahedral Gaussian grid [@Malardel2016], the octahedral
+Clenshaw-Curtis grid [@Hotta2018], and the HEALPix grid [@Gorski2005].
+Both SpeedyWeather.jl and its spherical harmonic transform `SpeedyTransforms` are also
+number format-flexible. Single-precision floating-point numbers
+(Float32) are the default as adopted by other modelling efforts [@Vana2017; @Nakano2018],
+but Float64 and other custom number formats can be used with a single
+code basis [@Klower2022; @Klower2020].
+Julia will compile to the choice of number format, the grid,
+and and other model components just-in-time. A simple parallelization
+across vertical layers is supported by Julia's multithreading.
+Output is stored as NetCDF files using
+[NCDatasets.jl](https://github.com/Alexander-Barth/NCDatasets.jl).
```

Comment:
```suggestion
NCDatasets.jl (reference to NCDatasets.jl).
```

Reply:
I don't think there is anything to cite for NCDatasets.jl @simone-silvestri, is there?

Reply:
maybe something like this?
```
@misc{ncdatasets,
  publisher = {GitHub},
  journal = {GitHub repository},
  howpublished = {\url{https://github.com/Alexander-Barth/NCDatasets.jl}},
  commit = {4f57d6a0e4c030202a07a60bc1bb1ed1544bf679}
}
```
I am not sure if there is a standard way to cite software but I wouldn't hide it in a hyperlink.

Reply:
OK, indeed! This is better!

Reply:
x-ref https://github.com/Alexander-Barth/NCDatasets.jl/issues/234

Reply:
Added!